### PR TITLE
feat: shared informer priority queue improvements

### PR DESCRIPTION
**Pull request by adityathebe**
* use queue for deleted objects as well. Get rid of delete buffers so every updates from the shared informers go into the queue.

### CODE REVIEW

File: scrapers/kubernetes/informers.go [hunk 1]
```diff
@@ -332,25 +325,64 @@ func kubeConfigIdentifier(ctx api.ScrapeContext) string {
...
+	}
+
+	if opResult := pqCompareKind(qa.Obj.GetKind(), qb.Obj.GetKind()); opResult != 0 {
+		return opResult
+	}
+
+	aLastUpdatedAt := *health.GetLastUpdatedTime(qa.Obj)
+	bLastUpdatedAt := *health.GetLastUpdatedTime(qb.Obj)
+
+	if aLastUpdatedAt.Before(bLastUpdatedAt) {
```
Comment (yashmehrotra) [inline scrapers/kubernetes/informers.go:354]:
```suggestion
	lastUpdatedTimeA := lo.FromPtr(health.GetLastUpdatedTime(qa.Obj))
	lastUpdatedTimeB := lo.FromPtr(health.GetLastUpdatedTime(qb.Obj))

	if lastUpdatedTimeB.Before(lastUpdatedTimeA) {
		return -1
	} else if lastUpdatedTimeA.Equal(lastUpdatedTimeB) {
		return 0
	} else {
		return 1
	}

```

Comment (yashmehrotra) [reply scrapers/kubernetes/informers.go:354]: B before A should return -1 as that signifies A is smaller and should be processed first right ? or is it the other way around ?

File: scrapers/incremental.go [hunk 2]
```diff
@@ -73,6 +73,11 @@ func ConsumeKubernetesWatchJobFunc(sc api.ScrapeContext, config v1.Kubernetes, q
 				}
 				obj := queueItem.Obj
 
+				if queueItem.Operation == "delete" {
```
Comment (yashmehrotra) [inline scrapers/incremental.go:76]: (nit) lets use constant vars for this

File: scrapers/kubernetes/informers.go (hunk 1 above)
Comment (adityathebe) [reply scrapers/kubernetes/informers.go:354]: > B before A should return -1 as that signifies A is smaller

`B` before `A` signifies `B` is smaller though.

**Pull request merged**